deploy/crds/openebs.io_blockdevices_crd.yaml

```diff
@@ -0,0 +1,247 @@
+                properties:
+                  compliance:
+                    description: Compliance is standards/specifications version implemented
+                      by device firmware  such as SPC-1, SPC-2, etc
+                    type: string
+                  deviceType:
+                    description: DeviceType represents the type of device like sparse,
+                      disk, partition, lvm, raid
+                    enum:
+                    - disk
+                    - partition
+                    - sparse
+                    - loop
+                    - lvm
+                    - raid
+                    type: string
+                  driveType:
+                    description: DriveType is the type of backing drive, HDD/SSD
+                    enum:
+                    - HDD
+                    - SDD
+                    - Unknown
+                    type: string
+                  firmwareRevision:
+                    description: FirmwareRevision is the disk firmware revision
+                    type: string
+                  hardwareSectorSize:
+                    description: HardwareSectorSize is the hardware sector size in
+                      bytes
+                    format: int32
+                    type: integer
+                  logicalBlockSize:
+                    description: LogicalBlockSize is the logical block size in bytes
+                      reported by /sys/class/block/sda/queue/logical_block_size
+                    format: int32
+                    type: integer
+                  model:
+                    description: Model is model of disk
+                    type: string
+                  physicalBlockSize:
+                    description: PhysicalBlockSize is the physical block size in bytes
+                      reported by /sys/class/block/sda/queue/physical_block_size
+                    format: int32
+                    type: integer
+                  serial:
+                    description: Serial is serial number of disk
+                    type: string
+                  vendor:
+                    description: Vendor is vendor of disk
+                    type: string
+                required:
+                - deviceType
+                - driveType
+                - hardwareSectorSize
+                - logicalBlockSize
+                - physicalBlockSize
+                type: object
+              devlinks:
+                description: DevLinks contains soft links of a block device like /dev/by-id/...
+                  /dev/by-uuid/...
+                items:
+                  description: DeviceDevLink holds the mapping between type and links
+                    like by-id type or by-path type link
+                  properties:
+                    kind:
+                      description: Kind is the type of link like by-id or by-path.
+                      enum:
+                      - by-id
+                      - by-path
```

Comment:
If I need to add a `by-uuid` or `by-partuuid` at a later time, should I modify the CRD also?

Reply:
Yes, but I am not sure if same version of CR(`v1alpha1`) should have varying fields across NDM versions. Can we consider using metadata/annotations until we're confident about the fields that NDM needs?

Reply:
The fields are not varying here. We are not adding any new fields. Its about adding one more value that is supported. cc: @prateekpandey14